code/postgres/src/main/scala/postgres/clients/PostgresClient.scala

```diff
@@ -65,52 +64,107 @@ trait UsingPostgres {
 
   implicit val dataSource = createDataSource
 
-  protected def write(partitionName: String, jsonRecords: List[JsonRecord])(implicit dataSource: DataSource) = {
+  def extractTableSchemaName(schemaAndTableName: String): (String, String) = {
+    if (schemaAndTableName.split("\\.").length == 2)
+      (schemaAndTableName.split("\\.")(0) , schemaAndTableName.split("\\.")(1))
+    else ("atomic","events")
+  }
+
 
+  protected def write(partitionName: String, jsonRecords: List[JsonRecord])(implicit dataSource: DataSource) = {
     val tableNames = Set(coreTableName)
     val tableNamesWithPartition =
       tableNames.map(tableName => (tableName, tableNameForPartition(tableName, partitionName))).toMap
+
     createParentTablesIfNotExists(tableNames)
     if (partitionName != null) { // partitioning is active
       createPartitionTablesIfNotExists(tableNamesWithPartition, partitionName)
-    }
-    withConnection { implicit connection =>
-      connection.setAutoCommit(false)
-      jsonRecords
-        .map { jsonRecord =>
-          createParentTableQueries(jsonRecord.json.values, tableNamesWithPartition)
+      withConnection { implicit connection =>
+        connection.setAutoCommit(false)
+        jsonRecords
+          .map { jsonRecord =>
+            createParentTableQueries(jsonRecord.json.extract[Map[String, Any]], tableNamesWithPartition)
```

Comment:
any particular reason why we changed the parameter in - 
createParentTableQueries(**jsonRecord.json.values**, tableNamesWithPartition)
 **to**
            createParentTableQueries(**jsonRecord.json.extract[Map[String, Any]]**, tableNamesWithPartition)